File: cascadingFailuresTaskMigration/src/main/java/input/Group.java
package input;

import lombok.Data;

import java.util.List;
import java.util.Set;

@Data
public class Group {
    private int groupId;
    private double groupLoad;
    private Robot leader;
    private Set<Integer> robotIdInGroup;
    private List<Task> assignedTasks;
    private double groupCapacity;
    private List<Robot> adLeaders;
}

File: cascadingFailuresTaskMigration_python/input/group.py
from dataclasses import dataclass, field
from typing import List, Set, Optional


@dataclass
class Group:
    group_id: int = 0
    group_load: float = 0.0
    leader: Optional['Robot'] = None
    robot_id_in_group: Set[int] = field(default_factory=set)
    assigned_tasks: List = field(default_factory=list)
    group_capacity: float = 0.0
    ad_leaders: List = field(default_factory=list)
